src/main/java/seedu/duke/task/TaskList.java

```diff
@@ -32,7 +35,43 @@ public void clearTaskList() {
      *
      * @param index the index of the task to be removed
      */
-    public void deleteTask(int index) {
-        taskList.remove(index);
+    public void deleteTask(int index) throws DukeException {
+        try {
+            taskList.remove(index);
+        } catch (IndexOutOfBoundsException exception) {
+            throw new DukeException(Message.ERROR_INDEX_OUT_OF_BOUNDS);
+        } catch (NumberFormatException exception) {
+            throw new DukeException(Message.ERROR_NUMBER_FORMAT);
+        }
+    }
+
+    public Task getTask(int index) throws DukeException {
+        try {
+            return taskList.get(index);
+        } catch (IndexOutOfBoundsException exception) {
+            throw new DukeException(Message.ERROR_INDEX_OUT_OF_BOUNDS);
+        } catch (NumberFormatException exception) {
+            throw new DukeException(Message.ERROR_NUMBER_FORMAT);
+        }
+    }
+
+    public void markTaskAsDone(int index) throws DukeException {
+        try {
+            taskList.get(index).setDone();
+        } catch (IndexOutOfBoundsException exception) {
+            throw new DukeException(Message.ERROR_INDEX_OUT_OF_BOUNDS);
+        } catch (NumberFormatException exception) {
+            throw new DukeException(Message.ERROR_NUMBER_FORMAT);
+        }
+    }
+
+    public TaskList findTasks(String keyword) {
+        TaskList matchingTaskList = new TaskList();
+        for (Task task: taskList) {

```

Comment:
Minor coding standard violation, more specifically the *white space within a statement*. So it should be written like this

```java
for (Task task : taskList)
```